### ISSUE TITLE
Allow building recipes outside haikuports tree

### ISSUE THREAD

**Issue by X547** [haikuports/haikuporter#197]
It would be nice to allow `haikuporter` use system package info instead of info inside `haikuports` tree. It will allow use recipe files for various projects outside `haikuports`.

**Comment by pulkomandy** [comment]
Just use the "package" tool directly if you are outside the tree. Cmake has support for it through cpack, for example.

What is the use of haikuporter in that situation?

**Comment by X547** [comment]
Many build systems can't build hpkg packages, for example Haiku makefile engine. haikuporter would simplify package generation.

> Cmake has support for it through cpack, for example.

Is more info about building `hpkg` by `cmake` available?

**Comment by pulkomandy** [comment]
You set cpack_binary_generator as explained in https://cmake.org/cmake/help/v3.12/module/CPack.html#variable:CPACK_GENERATOR and set the othervariables as documented there. Then use "make package"

**Comment by X547** [comment]
I can't see HPKG generator in cpack:
```
Generators
  7Z                           = 7-Zip file format
  DEB                          = Debian packages
  External                     = CPack External packages
  IFW                          = Qt Installer Framework
  NSIS                         = Null Soft Installer
  NSIS64                       = Null Soft Installer (64-bit)
  NuGet                        = NuGet packages
  STGZ                         = Self extracting Tar GZip compression
  TBZ2                         = Tar BZip2 compression
  TGZ                          = Tar GZip compression
  TXZ                          = Tar XZ compression
  TZ                           = Tar Compress compression
  TZST                         = Tar Zstandard compression
  ZIP                          = ZIP file format
```
Which generator should I use?
cmake version is 3.16.0.

**Comment by pulkomandy** [comment]
Then it seems my work to add support for the hpkg format has been removed. I had even submitted them upstream: https://cmake.org/Bug/view.php?id=14788

**Comment by mmuman** [comment]
Looks like they moved their bugtracker…
https://gitlab.kitware.com/cmake/cmake/issues/14788

**Comment by mmuman** [comment]
But I don't see CPackHpkg mentionned in the history…

**Comment by X547** [comment]
It it possible to include HPKG generator to haikuports patchset?

**Comment by pulkomandy** [comment]
Yes, and it would be great if we can ping cmake team to upstream it again, and if people would not remove my patches when they update cmake versions because "the new one compiles without the patches"

**Comment by X547** [comment]
I uploaded fixed patch for cmake 3.16.0: https://github.com/haikuports/haikuports/pull/4673.
